src/core/stream/adaptation/create_representation_estimator.ts

```diff
@@ -77,19 +89,35 @@ export default function createRepresentationEstimator(
                                        IABRRequestEndEvent>();
   const abrEvents$ = observableMerge(streamFeedback$, requestFeedback$);
 
-  /** Representations for which a `RepresentationStream` can be created. */
-  const playableRepresentations = adaptation.getPlayableRepresentations();
-  if (playableRepresentations.length <= 0) {
-    const noRepErr = new MediaError("NO_PLAYABLE_REPRESENTATION",
-                                    "No Representation in the chosen " +
-                                    "Adaptation can be played");
-    throw noRepErr;
-  }
+  const estimator$ = observableConcat(
+    observableOf(null), // Emit directly a first time on subscription
+    fromEvent(manifest, "decipherabilityUpdate") // then each time this event is triggered
+  ).pipe(
+    map(() : Representation[] => {
+      /** Representations for which a `RepresentationStream` can be created. */
+      const playableRepresentations = adaptation.getPlayableRepresentations();
+      if (playableRepresentations.length <= 0) {
+        const noRepErr = new MediaError("NO_PLAYABLE_REPRESENTATION",
+                                        "No Representation in the chosen " +
+                                        "Adaptation can be played");
+        throw noRepErr;
+      }
+      return playableRepresentations;
+    }),
+    distinctUntilChanged((prevRepr, newRepr) => {
+      if (prevRepr.length !== newRepr.length) {
+        return false;
+      }
+      for (let i = 0; i < newRepr.length; i++) {
+        if (prevRepr[i].id !== newRepr[i].id) {
+          return false;
+        }
+      }
+      return true;
+    }),
+    switchMap((playableRepresentations) =>
+      abrManager.get$(adaptation.type, playableRepresentations, clock$, abrEvents$)));
```

Comment:
Maybe we should in the future give an observable of playable representations to the get$ function. it seems more logical to me than re-calling it.